sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -68,20 +72,37 @@ export class AppConfigurationClient {
    * Initializes a new instance of the AppConfigurationClient class.
    * @param connectionString Connection string needed for a client to connect to Azure.
    */
-  constructor(connectionString: string) {
-    const regexMatch = connectionString.match(ConnectionStringRegex);
-    if (regexMatch) {
-      const appConfigCredential = new AppConfigCredential(regexMatch[2], regexMatch[3]);
-      this.client = new AppConfiguration(appConfigCredential, apiVersion, {
-        baseUri: regexMatch[1],
+  constructor(connectionString: string);
+  constructor(tokenCredential: TokenCredential, endpoint: string);
```

Comment:
I think the overload needs docs too.

Should the args be switched too? I think in event hubs we have something similar where host comes before tokenCredential. We should probably be consistent across packages where it makes sense to be.

Reply:
Yes and yes.